### PR TITLE
Fix @UserMessage does not work

### PR DESCRIPTION
**Pull request by Wadimz**
Fixes bug https://github.com/langchain4j/langchain4j/issues/694 
propagates userName from UserMessage where possible

## Summary by CodeRabbit

- **New Features**
	- Enhanced message processing to conditionally create or modify `UserMessage` objects based on the presence of a user's name, improving personalized interactions.
- **Tests**
	- Added and modified test cases to validate new message augmentation functionalities and content injection with usernames, ensuring reliability and expected behavior in user message processing.
- **Refactor**
	- Utilized utility functions for more efficient and cleaner checks on user message attributes, streamlining the handling of chat memory and user interactions.

### CODE REVIEW

File: langchain4j-core/src/main/java/dev/langchain4j/rag/content/injector/DefaultContentInjector.java [hunk 1]
```diff
@@ -79,6 +80,10 @@ public ChatMessage inject(List<Content> contents, ChatMessage chatMessage) {
         }
 
         Prompt prompt = createPrompt(chatMessage, contents);
+        if (chatMessage instanceof UserMessage && isNotNullOrBlank(((UserMessage)chatMessage).name())) {
+            prompt.toUserMessage(((UserMessage)chatMessage).name());
```
Comment (geoand) [inline langchain4j-core/src/main/java/dev/langchain4j/rag/content/injector/DefaultContentInjector.java]: I assume a `return` a missing here

Comment (Wadimz) [reply langchain4j-core/src/main/java/dev/langchain4j/rag/content/injector/DefaultContentInjector.java]: Absolutely, thanks for pointing, fixed!

**Review by geoand — APPROVED** [review]

File: langchain4j-core/src/test/java/dev/langchain4j/rag/DefaultRetrievalAugmentorTest.java [hunk 2]
```diff
@@ -136,7 +136,7 @@ void should_not_augment_when_router_does_not_return_retrievers(Executor executor
                 .executor(executor)
                 .build();
 
-        UserMessage userMessage = UserMessage.from("query");
+        UserMessage userMessage = UserMessage.from("userName", "query");
```
Comment (langchain4j) [inline langchain4j-core/src/test/java/dev/langchain4j/rag/DefaultRetrievalAugmentorTest.java]:
```suggestion
        UserMessage userMessage = UserMessage.from("query");
```

**Review by ghost — DISMISSED** [review]
@Wadimz thank you!

**Comment by scoooooott** [comment]
Thanks. I just found same problem and wanne see if there already a pull request.

**Pull request merged**